strudel-cli/src/strudel/main.py

```diff
@@ -1,145 +1,298 @@
 import typer
+import logging
 import subprocess
 import os
-import shutil
-import json
-import yaml
-from os.path import expanduser
+import traceback
 from typing import Optional
 from typing_extensions import Annotated
-from .utils import json_to_yaml, name_callback, version_callback, clear_cookiecutter_cache
+from .utils import (
+    json_to_yaml,
+    name_callback,
+    version_callback,
+    clear_cookiecutter_cache,
+)
 from enum import Enum
 from rich import print
 from rich.padding import Padding
+from rich.logging import RichHandler
+from rich.console import Console
 
 app = typer.Typer(no_args_is_help=True)
 
+logging.basicConfig(
+    level=logging.NOTSET, format="%(message)s", datefmt="[%X]", handlers=[RichHandler()]
+)
+_log = logging.getLogger(__name__)
+_console = Console()
+
+
+def _set_verbosity(vb):
+    if vb > 1:
+        _log.setLevel(logging.DEBUG)
+    elif vb > 0:
+        _log.setLevel(logging.INFO)
+    else:
+        _log.setLevel(logging.WARN)
+
+
+def _fatal_error(e):
+    _log.error(f"creating app: {e}")
+    _console.print_exception()
+
+def _clear_cache(action):
+    try:
+        clear_cookiecutter_cache()
+    except Exception as e:
+        _log.warning(f"could not clear cookiecutter cache: {e}")
+        print(f"[white]continuing to {action}...")
+
 
 @app.command()
 def create_app(
-  name: Annotated[str, typer.Argument(callback=name_callback, help="The name of your app. It's best to use only letters, hyphens, and underscores.")],
-  config: Annotated[str, typer.Option("--config", "-c", help="JSON file with configuration values to use to build your app.")] = "",
-  output_dir: Annotated[str, typer.Option("--output-dir", "-o", help="Directory where the app should be created. Defaults to current directory.")] = "",
-  branch: Annotated[str, typer.Option("--branch", "-b", help="Branch in strudel-kit repo that should be used for the templates. This option is primarily for use by contributors.")] = "main"
+    name: Annotated[
+        str,
+        typer.Argument(
+            callback=name_callback,
+            help="The name of your app. It's best to use only letters, hyphens, and underscores.",
+        ),
+    ],
+    config: Annotated[
+        str,
+        typer.Option(
+            "--config",
+            "-c",
+            help="JSON file with configuration values to use to build your app.",
+        ),
+    ] = "",
+    output_dir: Annotated[
+        str,
+        typer.Option(
+            "--output-dir",
+            "-o",
+            help="Directory where the app should be created. Defaults to current directory.",
+        ),
+    ] = "",
+    branch: Annotated[
+        str,
+        typer.Option(
+            "--branch",
+            "-b",
+            help="Branch in strudel-kit repo that should be used for the templates. This option is primarily for use by contributors.",
+        ),
+    ] = "main",
+    verbose: Annotated[int, typer.Option("--verbose", "-v", count=True)] = 0,
```

Comment:
What's the syntax for setting the verbosity level? I get an error if I do:
```
strudel create-app dan-test -v 1
```

Reply:
just -v, it's a flag. repeat it for more. This is the only way I've seen it done. Probably should repeat the option for the other command (if there's no way to share them with this system)